### ISSUE TITLE
SMPPaymentOptionCardReader ignored

### ISSUE THREAD

**Issue by wbison** [sumup/sumup-ios-sdk#97]
I want to skip the display of the reader/link screen when starting a payment. I am able to do this using the SumUp app by "Disabling Mobile payments" (a *very* confusing term btw, everything is mobile!). I thought I could achieve the same by using `SMPPaymentOptionCardReader` instead of `SMPPaymentOptionAny` as paymentOptions parameter but that has no effect: the screen where you can choose between reader of link is still shown.
(v3.4)

**Comment by pierceglennie** [comment]
I am facing this same issue. It was previously working fine (with no mobile payments option displayed) and I haven't made any changes or updated the SumUp SDK. Therefore, some server side change has presumably broken this.

I have "paymentOptions: .cardReader" within the checkout request. I'm using v3.4 of the SDK.

**Comment by Mike-Lopez** [comment]
Hi, 

Our apologies for the inconvenience.

I have raised a ticket with our developers and we are currently working on error you are facing.

I will update you immediately once it has been corrected.

**Comment by pierceglennie** [comment]
Thanks for the update

**Comment by pierceglennie** [comment]
Any update on this? Is it fixed with 4.0.0-beta.1? @mollidor

**Comment by WuDo** [comment]
+1

**Comment by mollidor** [comment]
Hi everyone,

we will deprecate the functionality to provide payment options when creating a checkout request in 4.0.0 which we will release today. Payment option provided to the user are governed by the merchant's settings only.
In the near future we will remove the option to pay by "Mobile Payments" via the SumUp SDK.
Appreciate the hint re naming, in the latest app and SDK 4.0.0 these will be called "Payment Links".

Kind regards,
Lukas

**Comment by pierceglennie** [comment]
Thanks for the update. Just to be clear, this means that soon it'll go back to the old behaviour of going straight to card screen since mobile payments is being removed from the SDK (and there will now be no option for it)? This would be a great improvement

**Comment by mollidor** [comment]
Exactly, with the card reader being the only payment option it will go straight to this screen. We'll keep you posted.

**Comment by pierceglennie** [comment]
Any idea when this update will go live @mollidor? We're about to start training people in our app and it's a bit frustrating having to tell them all to click through this option every since time

**Comment by pierceglennie** [comment]
Hi @mollidor, sorry to chase but is this still in the pipeline? Any idea how far away it is?

**Comment by danilkurkin1** [comment]
> Hi @mollidor. Is there any update on this issue? It's been a long time now since it was supposed to be getting fixed.

Can Report same behaviour and interesting to see the fix for it.

**Comment by pierceglennie** [comment]
I contacted SumUp by email on this and they said they don't monitor GitHub issues. Fortunately, there is a solution now for disabling payment links:

- Download and open the regular SumUp App
- Click on the icon "Home" on the lower left corner of the screen
- Select "Payment Links"
- Click on the settings icon on the top right corner
- Disable "Payment Links"

This will now disable them for the SDK also, assuming it's using the same login.
